packages/neuron-ui/src/components/GeneralSetting/LockWindowDialog/hooks.ts

```diff
@@ -0,0 +1,113 @@
+import { TFunction } from 'i18next'
+import { useCallback, useState } from 'react'
+import { verifyLockWindowPassword } from 'services/remote'
+import { updateLockWindowInfo, useDispatch } from 'states'
+import { isSuccessResponse } from 'utils'
+
+export const passwordLength = 4
+
+export const useOldPassword = ({ t }: { t: TFunction }) => {
+  const [oldPasswordErr, setOldPasswordErr] = useState('')
+  const [oldPassword, setOldPassword] = useState(new Array(passwordLength).fill(''))
+  const [verifySuccess, setVerifySuccess] = useState(false)
+  const onUpdateOldPassword = useCallback(
+    (v: string, idx: number) => {
+      const updatedOldPassword = oldPassword.toSpliced(idx, 1, v).join('')
```

Comment:
The API `toSpliced` is available in [Node@20](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/toSpliced), while the [Electron@0.28.0](https://github.com/nervosnetwork/neuron/blob/baee37ba9efd087457bb90f210374dab1d489444/packages/neuron-wallet/package.json#L96) is based on [Node@18](https://www.electronjs.org/docs/latest/tutorial/electron-timelines). Considering this, maybe we should avoid using such a new API to keep the main and renderer in the same API version

Reply:
Here is use the Chrome executor, https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/toSpliced from here we know Chrome 110 will start supporting it. https://releases.electronjs.org/ Electron's internal Chrome version is over 110.

Reply:
Any update on this conversation @homura

Reply:
> Any update on this conversation @homura 

It looks fine, the `toSplice` method now only runs in the runtime provided by Chrome now. BTW, it would be better to upgrade Electron to 30 to support more modern APIs, such as `toSplice` or `Intl`